### ISSUE TITLE
Getting float value on OS X

### ISSUE THREAD

**Issue by ilearnio** [kamicane/microseconds#1]
I'm not sure if it's related to the platform but on OS X it gives `145266269929.482` rather than  `145266269929482`. Is it a bug?

Related

``` js
now = μs.now()  // 147258820208.71

// ... some complex stuff

μs.since(now) // 10998.792022705078 (is that in milliseconds?)

// wrong parsing
μs.parse(now) // { microseconds: 209, milliseconds: 820, seconds: 18, minutes: 54, hours: 16, days: 1 }
```

**Comment by ilearnio** [comment]
`return +hr.join('')` instead of [`return (hr[0] * 1e9 + hr[1]) / 1e3`](https://github.com/kamicane/microseconds/blob/master/now.js#L11) fixes the issue

**Comment by kamicane** [comment]
That is normal, the result of `now()` and `since()` is in ~~milliseconds~~ microseconds

`hrtime` gives us nanosecond precision, which is then normalized to microseconds to be comparable with `performance.now` from browsers.

Typing manually in node.js (v5.4.0)

``` js
var then = μs.now()
var now = μs.since(then)
μs.parse(now).toString() //6 seconds 663 milliseconds 997 microseconds
```

The "timestamp" you get from .now() is just an arbitrary time in the past, this is primarly used to diff between two timestamps.

TLDR: getting floats is normal since the result is in microseconds, and `hrtime()` gives us nanosecond precision.

**Comment by ilearnio** [comment]
As you can see from my example, there is a wrong parsed output. It gives 209 microseconds instead of 871, and 1 day and 16 hours! I didn't launch this script for 1 day :)

**Comment by kamicane** [comment]
That's because you're parsing `now`, not the result of `since` (look at my example above). `hrtime` returns a timestamp at an arbitrary time in the past, while `Date.now()` (and my parsing of `performance.now()`) uses January 1 1970 00:00:00 UTC.

https://nodejs.org/api/process.html#process_process_hrtime

**Comment by ilearnio** [comment]
You were right. I have just tested it again like so:

``` js
var us = require('microseconds')
var now = us.now()

setTimeout(function() {
  var since = us.since(now)
  console.log('now ' + now)
  console.log('since ' + since)
  console.log('since parsed ' + us.parse(since))
},500)

// > now 254627129721.812
// > since 504968.2909851074
// > since parsed 504 milliseconds 968 microseconds
```

and it returns everything correctly. My bad. Sorry for bothering.
